packages/loader/container-loader/src/container.ts

```diff
@@ -748,7 +746,13 @@ export class Container
 		// Prefix all events in this file with container-loader
 		this.mc = loggerToMonitoringContext(ChildLogger.create(this.subLogger, "Container"));
 
-		this.options = cloneDeep(this.loader.services.options);
+		// NOTE: stringify + parse is usually bad practice. Here we deliberately chose to use it because we know the
+		// pieces we care about in the cloned object are primitive values, and it lets us avoid importing a full
+		// deep-clone library, so we keep the bundle size down. This code doesn't run a ton of times and if it's slower
+		// than an actual deep clone (improbable), the gain in network transfer time from a smaller bundle almost certainly
+		// outweighs that.
+		// Tracking alternative ways to handle this in AB#4129.
+		this.options = JSON.parse(JSON.stringify(this.loader.services.options));
```

Comment:
have you seen this?
https://developer.mozilla.org/en-US/docs/Web/API/structuredClone

Could potentailly do something like: 
```
		this.options =
			structuredClone !== undefined
				? structuredClone(this.loader.services.options)
				: JSON.parse(JSON.stringify(this.loader.services.options));

```

Reply:
maybe just track [structuredClone](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone) for now. it also has a polyfil we could consider requiring. overall seems better than lodash, as it might be there by default

Reply:
Nice! It's only available in node starting with v17 though?

Reply:
Ah, already considered in your snippet 😄

Reply:
That's awesome!  I was definitely pondering whether this is something JS runtime should include.